examples/ruby/app.rb

```diff
@@ -0,0 +1,209 @@
+public_key = private_key.public_key
+
+# -----------------------------------
+# BUILD THE CLIENT ASSERTION
+# -----------------------------------
+
+header = {
+  'alg' => 'RS384',
+  'kid' => KID,
+  'typ' => 'JWT'
+}
+puts "HEADER:\n#{header.to_json}"
+
+payload = {
+  'iss' => CLIENT_ID,
+  'sub' => CLIENT_ID,
+  'aud' => TOKEN_URL,
+  'exp' => Time.now.to_i + (5 * 60),
+  'jti' => 1,
+}
+puts "PAYLOAD:\n#{payload.to_json}"
+puts '---------------------'
+
+# token = JWT.encode payload, key, algorithm, header_fields
+client_assertion = JWT.encode payload, private_key, algorithm='RS384', header_fields=header
+puts "CLIENT ASSERTION:\n#{client_assertion}"
+puts '---------------------'
+
+# -----------------------------------
+# BUILD THE ACCESS TOKEN REQUEST
+# -----------------------------------
+parameters = {
+  'client_id' => CLIENT_ID,
+  'client_assertion' => client_assertion,
+  'client_assertion_type' => 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer',
+  'grant_type' => 'client_credentials',
+  'scope' => 'system/*.*'
+}
+puts "PARAMETERS:\n#{parameters.to_json}"
+puts '---------------------'
+
+decoded_assertion = JWT.decode client_assertion, public_key, true, { algorithm:'RS384' }
+puts "DECODED:\n#{decoded_assertion.to_json}"
+puts '---------------------'
+
+# -----------------------------------
+# REQUEST AN ACCESS TOKEN
+# -----------------------------------
+url = URI(TOKEN_URL)
+http = Net::HTTP.new(url.host, url.port);
+request = Net::HTTP::Post.new(url)
+request['Content-Type'] = 'application/json'
+request.body = parameters.to_json
+response = http.request(request)
+body = response.read_body
+access_token = JSON.parse(body)
+puts "ACCESS TOKEN:\n#{body}"
+puts "ACCESS TOKEN:\n#{access_token['access_token']}"
+puts '---------------------'
+
+# -----------------------------------
+# FHIR: Get a Patient
+# -----------------------------------
+url = URI("http://localhost:3000/fhir/r4/Patient/1")
+http = Net::HTTP.new(url.host, url.port);
+request = Net::HTTP::Get.new(url)
+request['Accept'] = 'application/fhir+json'
+request['Authorization'] = "Bearer #{access_token['access_token']}"
+response = http.request(request)
+body = response.read_body
+
+puts "GET FHIR PATIENT:\n#{body}"
+puts '---------------------'
+
+# -----------------------------------
+# FHIR: Get some Observations
+# -----------------------------------
+url = URI("http://localhost:3000/fhir/r4/Observation?patient=1")
+http = Net::HTTP.new(url.host, url.port);
+request = Net::HTTP::Get.new(url)
+request['Content-Type'] = 'application/fhir+json'
+request['Accept'] = 'application/fhir+json'
+request['Authorization'] = "Bearer #{access_token['access_token']}"
+response = http.request(request)
+body = response.read_body
+json = JSON.parse(body)
+puts "GET FHIR OBSERVATIONS FOR PATIENT:\nCOUNT: #{json['total']}"
+puts '---------------------'
+
+# -----------------------------------
+# FHIR: Write an Observation
```

Comment:
Maybe exclude this section from here for now until we do support this?

Reply:
I thought about that... but then again, I thought it might be useful for potential users to be told that it does not work while they are staring at the code. Either way, I am fine. Let me know.

Reply:
Yeah my thought was we show the ones that are writable in the API and I thought this was kind of pointing out "Hey we don't support this yet!" which feels a bit bad (even though we are planning to).

Reply:
I actually do like the idea of leaving it in - if I was reading this code fresh I would appreciate seeing as many examples/details as possible, even if something is not yet supported. Sort of nice because it helps frame what will be possible, and affects how I might plan to use the API long term.

Reply:
Haha I would appear to be overruled, so we can go with the majority.